crypto/signer.go

```diff
@@ -35,10 +35,20 @@ var sigpbHashLookup = map[crypto.Hash]sigpb.DigitallySigned_HashAlgorithm{
 // Signer is responsible for signing log-related data and producing the appropriate
 // application specific signature objects.
 type Signer struct {
+	KeyID  int64
 	Hash   crypto.Hash
 	Signer crypto.Signer
 }
 
+// NewSigner returns a new annotated signer.
+func NewSigner(LogID int64, signer crypto.Signer, hash crypto.Hash) *Signer {
+	return &Signer{
+		KeyID:  LogID,
+		Hash:   hash,
+		Signer: signer,
+	}
+}
+
 // NewSHA256Signer creates a new SHA256 based Signer.
```

Comment:
Is this function going to stick around?  

If it is, then it should probably get some comment to indicate that the resulting signed isn't associated with a particular key ID. Also, the changes to various `..._test.go` files might not be needed.

Reply:
I would like to remove `NewSha256Signer`.  I've got a PR queued up for the `certificatetransparency-go` repo, but obviously this PR needs to be merged first.

Reply:
I'll add a TODO to remove it.